cmd/util/cmd/epoch-recovery/cmd/reset.go

```diff
@@ -0,0 +1,192 @@
+	}
+	if !exists {
+		log.Fatal().Str("path", snapshotPath).Msgf("root-protocol-snapshot.json file does not exists in the `boot-dir` given")
+	}
+
+	// read root protocol-snapshot.json
+	bz, err := io.ReadFile(snapshotPath)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("could not read root snapshot file")
+	}
+
+	// unmarshal bytes to inmem protocol snapshot
+	snapshot, err := convert.BytesToInmemSnapshot(bz)
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not convert array of bytes to snapshot")
+	}
+
+	// extract arguments from reset epoch tx from snapshot
+	txArgs := extractResetEpochArgs(snapshot)
+
+	// ancode to JSON
+	enc := encodeArgs(txArgs)
+
+	// write JSON args to file
+	err = io.WriteFile(argsPath, enc)
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not write jsoncdc encoded arguments")
+	}
+}
+
+// extractResetEpochArgs extracts the required transaction arguments for the `resetEpoch` transaction
+func extractResetEpochArgs(snapshot *inmem.Snapshot) []cadence.Value {
+
+	// get current epoch
+	epoch := snapshot.Epochs().Current()
+
+	// read random source from epoch
+	randomSource, err := epoch.RandomSource()
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not get random source from epoch")
+	}
+
+	// read first view
+	firstView, err := epoch.FirstView()
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not get first view from epoch")
+	}
+
+	// read final view
+	finalView, err := epoch.FinalView()
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not get final view from epoch")
+	}
+
+	return convertResetEpochArgs(randomSource, flagPayout, firstView, finalView)
+}
+
+// convertResetEpochArgs converts the arguments required by `resetEpoch` to cadence representations
+func convertResetEpochArgs(randomSource []byte, payout, firstView, finalView uint64) []cadence.Value {
+
+	args := make([]cadence.Value, 0)
+
+	// add random source
+	args = append(args, cadence.NewString(hex.EncodeToString(randomSource)))
+
+	// add payout
+	var cdcPayout cadence.Value
+	var err error
+
+	if payout != 0 {
+		cdcPayout, err = cadence.NewUFix64(fmt.Sprintf("%d.0", payout))
+		if err != nil {
+			log.Fatal().Err(err).Msg("could not convert payout to cadence type")
+		}
+	} else {
+		cdcPayout = cadence.NewOptional(nil)
+	}
+	args = append(args, cdcPayout)
+
+	// add first view
+	args = append(args, cadence.NewUInt64(firstView))
+
+	// add final view
+	args = append(args, cadence.NewUInt64(finalView))
+
+	return args
+}
+
+// encodeArgs JSON encodes `resetEpoch` transaction arguments
+func encodeArgs(args []cadence.Value) []byte {
+
+	arguments := make([]interface{}, 0, len(args))
+
+	for index, cdcVal := range args {
+
+		encoded, err := jsoncdc.Encode(cdcVal)
+		if err != nil {
+			log.Fatal().Err(err).Msg("could not encode cadence arguments")
+		}
+
+		var arg interface{}
+		err = json.Unmarshal(encoded, &arg)
+		if err != nil {
+			log.Fatal().Err(err).Msg("could not unmarshal cadence arguments")
+		}
+
+		arguments[index] = arg
```

Comment:
⚠️  This will cause an index out of bounds panic, because `arguments` is initialized with 0 length and is never appended to.